torch/_dynamo/variables/misc.py

```diff
@@ -285,22 +380,34 @@ def call_method(
         args: "List[VariableTracker]",
         kwargs: "Dict[str, VariableTracker]",
     ) -> "VariableTracker":
-        if name != "save_for_backward":
+        if name not in ("saved_tensors", "save_for_backward"):
             unimplemented(f"autograd.Function context method: {name}")
 
         if not self.inference:
             assert self.source and not kwargs
             tx.output.side_effects.track_save_for_backward(self, args)
 
-        return variables.ConstantVariable(
-            None, **VariableTracker.propagate(self, args, kwargs.values())
-        )
+        options = VariableTracker.propagate(self, args, kwargs.values())
+        if name == "saved_tensors":
+            return variables.TupleVariable(list(self._saved_tensors), **options)
+
+        assert name == "save_for_backward"
+        if not hasattr(self, "_saved_tensors"):
+            self._saved_tensors = []
+        for arg in args:
+            # as_proxy can return constant values or other non proxy values
+            if isinstance(arg.as_proxy(), torch.fx.Proxy):
+                arg.as_proxy().node.meta["whitelisted"] = True
+            self._saved_tensors.append(arg)
```

Comment:
Do we need to whitelist things saved directly to the ctx object? Or is that immediately a graph break during the forward?

Reply:
Sorry, what do you mean?

Reply:
If we have an autograd.Function that looks like:
```
class Func(autograd.Function):
    @staticmethod
    def forward(ctx, x):
        ctx.x = x  # Well, we don't recommend users save tensors directly
        return x.view(-1)
    @staticmethod
    def backward(ctx, grads)
        return grads.view(ctx.x.shape)
        
class Func(autograd.Function):
    @staticmethod
    def forward(ctx, x):
        ctx.x_shape = x.shape
        return x.view(-1)
    @staticmethod
    def backward(ctx, grads)
        return grads.view(ctx.x_shape)

```

Do we need to whitelist `ctx.x_shape` or `ctx.x`? `ctx.x_shape` is not a Tensor, so it probably doesn't get a Proxy?